### PR TITLE
Add Cardknox Gateway

### PR DESCRIPTION
**Pull request by achasveachas**

### CODE REVIEW

File: app/models/spree/gateway/cardknox.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        ActiveMerchant::Billing::CardknoxGateway
+      end
+    end
+end
+
+# A hack because ActiveMerchant's CardknoxGateway class returned an error when a paymentmethod was submitted that didn't respond to track_data and manual_entry methods
+# Can be removed when Pull Request #2580 is merged to ActiveMerchant
+
+module AdditionalMethods
+    attr_accessor :track_data, :manual_entry
```
Comment (jhawthorn) [inline app/models/spree/gateway/cardknox.rb]: I don't want to add anything to solidus_gateway which monkey patches solidus_core. Maybe this gateway should instead be in its own extension.

Comment (tvdeyen) [reply app/models/spree/gateway/cardknox.rb]: Valid point, John. I think we should start to encourage people to write separate extensions for payment providers in stead of bloating this (already arguable) gem 👌🏻

Comment (achasveachas) [reply app/models/spree/gateway/cardknox.rb]: Hmmm I was going to remove the monkey patch and just overwrite the offending methods myself. But if you'd prefer not to add more gateways I can look into creating a new extension...

**Comment by achasveachas** [comment]
I got all the tests passing for all of the builds (the one where it's failing seems to be for an unrelated reason?)

I didn't get rid of the monkey patch, and in light of the discussion above am not sure if it's worth working on it.

File: app/models/spree/gateway/cardknox.rb (hunk 1 above)
Comment (tvdeyen) [reply app/models/spree/gateway/cardknox.rb]: I think an extension on its own would be better.

You could then add you own payment source class that inherits from `Spree::PaymentSource` with your extra methods added.

Also you don't have to deal with legacy Solidus versions and old Active Merchant versions.

**Pull request closed without merging**